connectors/sources/jira.py

```diff
@@ -887,17 +887,25 @@ async def _put_issue(self, issue):
         async for issue_metadata in self.jira_client.get_issues_for_issue_key(
             key=issue.get("key")
         ):
-            response_fields = {
-                self.custom_fields.get(k, k): v
-                for k, v in issue_metadata.get("fields").items()
-            }
+            response_custom_fields = {}
+            response_fields = copy(issue_metadata.get("fields"))
+            for k, v in response_fields.items():
+                if self.custom_fields.get(k):
+                    response_custom_fields[self.custom_fields[k]] = v
+
+            for k in self.custom_fields.keys():
+                if k in response_fields:
+                    del response_fields[k]
```

Comment:
Yes, in that case we need both fields. So, would it be okay to have a duplicate fields in the document because the community member wants custom fields separately https://github.com/elastic/connectors/issues/2415#issuecomment-2093077220?

Reply:
Perhaps I'm misunderstanding. I'd expect that the likelyhood of duplicate fields is low. But is that wrong? is `self.custom_fields` always a subset of `response_fields`? If _all_ of the fields in `self.custom_fields` are always in `response_fields`, then I think your current logic is good. But if there's just a chance that sometimes there are duplicates between the two, I don't think we should remove the original fields, in case they have a value different from the custom fields.

Reply:
Yes, `self.custom_fields` will always be a subset of `response_fields`.

Reply:
Ah ok, then I'd originally misunderstood. Can you revert back to using `del response_fields[k]`?

Reply:
I haven't made any changes yet since i was waiting for the conclusion on this part 👍